### PR TITLE
Added required PHP modules to composer.json

### PR DESCRIPTION
**Pull request by fballiano**
Everything is explained in ISSUE https://github.com/OpenMage/magento-lts/issues/2017

### CODE REVIEW

**Comment by fballiano** [comment]
@tmotyl could you help me updating the composer lock?

**Comment by tmotyl** [comment]
`composer update —lock` should be enough

**Comment by fballiano** [comment]
to be able to run composer update --lock I had to update the composer.json file again adding a few things, do you think it's ok?

**Review by Sdfendor — APPROVED** [review]

**Comment by tmotyl** [comment]
I can't say whether the change of ""magento-root-dir": "."" is correct or not, I'm not sure what it does.

**Comment by fballiano** [comment]
> I can't say whether the change of ""magento-root-dir": "."" is correct or not, I'm not sure what it does.

it seems to be where composer installs magento, it's weird but without that any composer command wouldn't work

**Comment by luigifab** [comment]
I think that the good composer command is: `composer update --lock --no-install`
With it, `magento-root-dir` changes are not required.

**Comment by fballiano** [comment]
Did as @luigifab and worked flawlessly, thank you so much! :-)

**Comment by luigifab** [comment]
If we add [that](https://github.com/OpenMage/magento-lts/issues/2017#issuecomment-1048620586), peoples need to install **php-ftp** and **php-iconv**, right?
I never installed them, and I don't want to install them (ok I use `--ignore-platform-reqs`, but...).

**Comment by fballiano** [comment]
> If we add [that](https://github.com/OpenMage/magento-lts/issues/2017#issuecomment-1048620586), peoples need to install **php-ftp** and **php-iconv**, right? I never installed them, and I don't want to install them (ok I use `--ignore-platform-reqs`, but...).

I think so. bBut you sent me the list of extensions :-)

**Comment by luigifab** [comment]
Remove ext-ftp and ext-iconv, zlib? I don't know, and let's go.

**Comment by fballiano** [comment]
mmm but it seems to me that the ConfigurableSwatches module uses iconv and it is required by app/code/core/Mage/Install/etc/install.xml

File: composer.json [hunk 1]
```diff
@@ -46,7 +58,6 @@
   "extra": {
     "branch-alias": {
       "dev-main": "1.9.4.x-dev"
-    },
```
Comment (Flyingmana) [inline composer.json:49]: unrelated change which needs to be avoided

Comment (fballiano) [reply composer.json]: removed

**Comment by Flyingmana** [comment]
so following https://magento.stackexchange.com/questions/61377/minimum-number-of-php-modules-needed-to-run-magento it seems we got all the commonly needed exceptions.
Only ftp seems to not be commonly needed, so if we remove this one I think the PR is ready.

Thanks for all the work already :)

**Comment by fballiano** [comment]
ftp extension removed

**Review by ADDISON74 — APPROVED** [review]

**Review by Flyingmana — APPROVED** [review]
